csharp/src/Ice/Discovery/Locator.cs

```diff
@@ -165,16 +166,17 @@ internal Locator(ILocatorRegistryPrx registry, ILookupPrx lookup, ObjectAdapter
             // of the lookup proxy.
 
             // Dummy proxy for replies which can have multiple endpoints (but see below).
-            IObjectPrx lookupReply = _replyAdapter.CreateProxy(
-                "dummy",
-                IObjectPrx.Factory).Clone(invocationMode: InvocationMode.Datagram);
+            IObjectPrx lookupReply = _replyAdapter.CreateProxy("dummy", IObjectPrx.Factory);
+            Debug.Assert(lookupReply.InvocationMode == InvocationMode.Datagram);
 
-            foreach (Endpoint endpoint in lookup.Endpoints)
+            foreach (Endpoint endpoint in _lookup.Endpoints)
             {
-                // lookup's invocation mode is Datagram
-                Debug.Assert(endpoint.Transport == Transport.UDP);
+                if (!endpoint.IsDatagram)
+                {
+                    throw new InvalidConfigurationException("lookup can only have udp endpoints");
```

Comment:
Should we reference the property here so the user isn't confused at to what "lookup" means?

Reply:
Fixed.